runtime/src/bank.rs

```diff
@@ -4003,7 +4003,25 @@ impl Bank {
     }
 
     pub fn get_account_modified_slot(&self, pubkey: &Pubkey) -> Option<(AccountSharedData, Slot)> {
-        self.rc.accounts.load_slow(&self.ancestors, pubkey)
+        self.load_slow(&self.ancestors, pubkey)
+    }
+
+    fn load_slow(
+        &self,
+        ancestors: &Ancestors,
+        pubkey: &Pubkey,
+    ) -> Option<(AccountSharedData, Slot)> {
+        // get_account (= this fn caller) can be called for various on-consensus
+        // code like inflation, feature activation and native program loading
+        // before freezing and it's generally impossible to make static/lexical guarantee of RPC or
+        // on-consensus.. so just pass !freeze_started for LoadHint::FixedMaxRoot as a close
+        // approximation of replaying/banking.
+        let load_hint = if !self.freeze_started() {
+            LoadHint::FixedMaxRoot
+        } else {
+            LoadHint::Unspecified
+        };
```

Comment:
I fell like this this might not be the most intuitive mapping to what each `LoadHint` does. The purpose of the load hint is just this assert here: https://github.com/solana-labs/solana/blob/b8ced7f7334a4e2f2558affd2b461c456f445a6b/runtime/src/accounts_db.rs#L2522 right?

For instance, if a different thread than replay keeps around a reference to an unfrozen bank and calls load, they may see more than one failure if the root keeps moving, triggering that assert.

I think the safest thing to do here is to default to the same behavior `LoadHint::Unspecified`, but in specific paths like inflation, feature activation and native program loading, plumb the `LoadHint::FixedMaxRoot` where we know it's safe.

Reply:
@carllin thanks for raising that concern. Yeah, let's be on the safer side. I've went through all uses of `get_account`: 0465e355ee196c2c1fe330eff6dc87b85389070a. Maybe this is what you wanted?

fyi, this is much simpler version of https://github.com/solana-labs/solana/commit/77010dbbe2f2f2cd73b228bf5a0e86f90632eac4. ;)

Reply:
Awesome, looks great 😄